app/controllers/api/v2/hosts_controller.rb

```diff
@@ -170,6 +182,32 @@ def facts
 
       private
 
+      def merge_interfaces(host)
+        merge = InterfaceMerge.new
+        merge.run(host.interfaces, host.compute_resource.try(:compute_profile_for, host.compute_profile_id))
+      end
+
+      def host_attributes(params)
+        return {} if params.nil?
+
```

Comment:
Could we deep_clone params here? We shouldn't change params values in this method, we should just return new hash with correctly nested values I think. The reason is that something else might expect params in their original form...